app/controllers/uploads_controller.rb

```diff
@@ -12,22 +10,29 @@ def new
     @upload = new_upload(params[:csv_type])
     return if @upload.csv_type_check?
 
-    log_and_display_error(errors_for_alert([@upload]), 'Warning')
+    alert_and_log(@upload.errors.full_messages.join(', '))
     redirect_to dashboards_path
   end
 
   def create
     @upload = Upload.create(merged_params)
 
     begin
-      data = load_csv
-      display_failed_instances(data.failed_instances)
+      failed = load_csv.failed_instances
+
+      @upload.check_for_headers
+      headers = @upload.all_warnings
```

Comment:
So i might be responsible for this one. Perhaps we could make an alias on the 'all_warnings' 

The idea was to create a new "warnings" or "notices" handler that inherits from ActiveModel::Errors to cleanup some of the mess that was "Flashable."

I proposed calling it "warnings" because its equally generic to "errors," perhaps for this specific model we could assign an alias "header_warnings" but I do see value in keeping this concept "warnings" or "notices" (errors that are not actual errors preventing persistence of object) from becoming too specific to the context they are being user for.